### PR TITLE
defined attachment types

### PR DESCRIPTION
**Pull request by YKuvonchbek**

### CODE REVIEW

File: app/src/main/java/co/tinode/tindroid/MessagesFragment.java [hunk 1]
```diff
@@ -10,7 +10,7 @@
 import android.database.Cursor;
 import android.graphics.Bitmap;
 import android.graphics.BitmapFactory;
-import androidx.exifinterface.media.ExifInterface;
+import android.media.ExifInterface;
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/MessagesFragment.java]: From https://developer.android.com/reference/android/media/ExifInterface

<i>Note: It is recommended to use the AndroidX ExifInterface Library since it is a superset of this class. </i>

Why do you want to make this change?

File: app/src/main/java/co/tinode/tindroid/UiUtils.java [hunk 2]
```diff
@@ -19,6 +19,7 @@
 import android.graphics.drawable.AnimationDrawable;
 import android.graphics.drawable.Drawable;
 import android.graphics.drawable.LayerDrawable;
+import android.media.ExifInterface;
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/UiUtils.java]: The same question: why are you replacing androidx.exifinterface.media.ExifInterface with android.media.ExifInterface?

File: app/src/main/java/co/tinode/tindroid/media/AttachmentType.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
+package co.tinode.tindroid.media;
+
+/**
+ * Attachment types
+ */
+
+public class AttachmentType {
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/media/AttachmentType.java]: If you define constants, why not use `Enum` instead of static class members?

File: app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java [hunk 4]
```diff
@@ -94,6 +95,51 @@ public static boolean hasClickableSpans(final Drafty content) {
         return false;
     }
 
+    /**
+     * Determine attachment type for presentation on UI
+     */
+    public static String getAttachmentType(final Drafty content) {
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java]: This returns the type of the first entity which may not be the attachment. The name of the method is misleading.

File: app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java [hunk 5]
```diff
@@ -271,7 +317,8 @@ public void onClick(View widget) {
                         }, content);
                     } catch (ClassCastException | NullPointerException ignored) {}
                     break;
-                case "MN": break;
+                case "MN":
+                    break;
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java]: This seems to be an unnecessary change. If you want to just reformat the code then please do it in a separate PR.

File: app/src/main/res/drawable/ic_mic_none_white_24dp.xml [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (or-else) [inline app/src/main/res/drawable/ic_mic_none_white_24dp.xml]: The name of the file is misleading. It's has `white` in the name when the color is not white.

File: app/src/main/res/drawable/ic_send.xml [hunk 7]
```diff
@@ -4,6 +4,6 @@
         android:viewportWidth="24.0"
         android:viewportHeight="24.0">
     <path
-        android:fillColor="#dd000000"
+        android:fillColor="@color/colorAccent"
```
Comment (or-else) [inline app/src/main/res/drawable/ic_send.xml]: Older versions of Android have a problem with using color references in drawables.

File: app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java (hunk 4 above)
Comment (or-else) [reply app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java]: This method appears to be unused. Why is it needed?

File: app/src/main/java/co/tinode/tindroid/MessagesFragment.java [hunk 8]
```diff
@@ -690,6 +702,14 @@ private void sendText() {
         }
     }
 
+    // Send video as a link
+    private static Drafty draftyVideo(String mimeType, byte[] bits, int width, int height, String fname,
```
Comment (or-else) [inline app/src/main/java/co/tinode/tindroid/MessagesFragment.java]: This method appears to be unused. Why do you need it?

File: app/src/main/java/co/tinode/tindroid/MessagesFragment.java (hunk 1 above)
Comment (Kuvonchbeky) [reply app/src/main/java/co/tinode/tindroid/MessagesFragment.java]: Because androidx.exifinterface.media.ExifInterface not found

File: app/src/main/java/co/tinode/tindroid/media/AttachmentType.java (hunk 3 above)
Comment (Kuvonchbeky) [reply app/src/main/java/co/tinode/tindroid/media/AttachmentType.java]: It's just habit, I will use `Enum`

File: app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java (hunk 4 above)
Comment (Kuvonchbeky) [reply app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java]: In the future for presenting different kind of attachments (i. e. video, audio, pdf document and etc)

File: app/src/main/java/co/tinode/tindroid/MessagesFragment.java (hunk 8 above)
Comment (Kuvonchbeky) [reply app/src/main/java/co/tinode/tindroid/MessagesFragment.java]: For video sharing feature

**Comment by or-else** [comment]
Overall this PR seems to be dependent on #72

Please follow these rules:

One feature - one PR.

Not two features - one PR. Not one feature - two PRs.

One feature - one PR.

Thanks!

File: app/src/main/java/co/tinode/tindroid/MessagesFragment.java (hunk 8 above)
Comment (or-else) [reply app/src/main/java/co/tinode/tindroid/MessagesFragment.java]: Yes, I understand that it's for video, but it's currently unused. It also has nothing to do with mentions that you otherwise work on. Please keep it simple:

* All code related to a feature should be in one PR. The PR should compile and work as expected.
* Code not related to the feature should not be in the PR. If it's not related to the feature keep it out. Send it separately in another PR.

File: app/src/main/java/co/tinode/tindroid/MessagesFragment.java (hunk 1 above)
Comment (or-else) [reply app/src/main/java/co/tinode/tindroid/MessagesFragment.java]: Then you are doing something wrong. Please fix your build environment.

File: app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java (hunk 4 above)
Comment (or-else) [reply app/src/main/java/co/tinode/tindroid/media/SpanFormatter.java]: Please keep it simple. Add only the code needed for the feature you are adding.
